services/offline/src/lib/cacheable-section-state.js

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import {
+    createStore,
+    useGlobalState,
+    useGlobalStateMutation,
+    GlobalStateProvider,
+} from './global-state-service'
+import { useOfflineInterface } from './offline-interface'
+
+// Functions in here use the global state service to manage cacheable section
+// state in a performant way
+
+/**
+ * Create a store for Cacheable Section state.
+ * Expected to be used in app adapter
+ */
+export function createCacheableSectionStore() {
+    const initialState = { recordingStates: {}, cachedSections: {} }
+    return createStore(initialState)
+}
+
+/**
+ * Provides context for a global state context which will track cached
+ * sections' status and cacheable sections' recording states, which will
+ * determine how that component will render. The provider will be a part of
+ * the OfflineProvider.
+ */
+export function CacheableSectionProvider({ children, store }) {
```

Comment:
One reason I can see for having it this way is to make testing easier, but in that case having an `initialState` prop instead of `store` would be more ergonomic.

Reply:
I guess my (faulty) reasoning was to follow the same 'no initialization in the app runtime' rule that kept the offline interface in the adapter instead of the runtime, but I guess since there are no side effects to the store, it's fine to keep internal!

Reply:
We do have initialization and state in app runtime currently (i.e. the engine is initialized by the data service and the alerts manager by the alerts service).  We probably do want in the future to support a kind of "delegating provider" which would allow you to pass in externally-sourced initialized state, but for the main provider side-effect-free state should be fine 👍